docs/source/documentation/platform/infrastructure/terraform.html.md.erb

```diff
@@ -0,0 +1,219 @@
+---
+owner_slack: "#data-platform-notifications"
+title: Terraform
+last_reviewed_on: 2023-07-25
+review_in: 6 months
+---
+
+# <%= current_page.data.title %>
+
+Data Platform's Core Infrastructure team offer a standardised pipeline for running Terraform. This is done via GitHub Actions and is available to all teams.
+
+Terraform components are scanned by [Checkov](https://github.com/bridgecrewio/checkov) and [tfsec](https://github.com/aquasecurity/tfsec)
+
+Terraform components are also linted by the Super Linter with [fmt](https://developer.hashicorp.com/terraform/cli/commands/fmt), [tflint](https://github.com/terraform-linters/tflint) and [terrascan](https://github.com/accurics/terrascan)
+
+## Style guide
+
+### Structure
+
+* data sources should be placed in `data.tf`
+
+* providers and versions should be placed in `terraform.tf`
+
+* variables should be placed in `variables.tf`
+
+* outputs should be placed in `outputs.tf`
+
+* resources should be grouped in files which are based from their name, e.g.
+
+  * IAM policies should be placed in `iam-policies.tf`
+
+  * IAM roles should be placed in `iam-roles.tf`
+
+  * KMS keys should be placed in `kms-keys.tf`
+
+* locals should be placed in `locals.tf`
+
+* modules should be placed in `modules/`
+
+### Naming
+
+* resources should be named using snake case, e.g. `resource "aws_iam_role" "example_role_name" {}`
+
+* resources shouldn't contain what they are, e.g. `resource "aws_iam_role" "example_role_name" {}` is preferred over `resource "aws_iam_role" "example_role_name_role" {}`
+
+* modules should be named using snake case, e.g. `module "example_module_name" {}`
+
+* modules should be name relative to what they are doing, e.g. `module "example_iam_role" {}`
+
+## Creating a new Terraform component
+
+To create a new Terraform component, you will need to:
+
+1. Create a directory in `terraform`, e.g. `terraform/aws/${AWS_ACCOUNT_NAME}/${COMPONENT}`
+
+1. Add the following files substituting the values in `${}` as appropriate
+
+    | Key | Value | Example |
+    |---|---|---|
+    | `AWS_ACCOUNT_NAME` | Name of the account as it appears in<br />AWS IAM Identity Center | `analytical-platform-development` |
+    | `AWS_ACCOUNT_ID` | ID of the account as it appears in<br />AWS IAM Identity Center | `123456789012` |
+    | `COMPONENT` | Name of the component<br />(We refer to a **component** as a collection of resources that create a service) | `eks` |
+    | `ENVIRONMENT` | Name of the environment<br /> | `development` |
+    | `IS_PRODUCTION` | Whether the environment is production<br /> | `false` |
+
+    `data.tf`
```

Comment:
```suggestion
Here is a table of examples of keys/values used in defining terraform assets. These can be used in your files using `${}` as appropriate:

    | Key | Value | Example |
    |---|---|---|
    | `AWS_ACCOUNT_NAME` | Name of the account as it appears in<br />AWS IAM Identity Center | `analytical-platform-development` |
    | `AWS_ACCOUNT_ID` | ID of the account as it appears in<br />AWS IAM Identity Center | `123456789012` |
    | `COMPONENT` | Name of the component<br />(We refer to a **component** as a collection of resources that create a service) | `eks` |
    | `ENVIRONMENT` | Name of the environment<br /> | `development` |
    | `IS_PRODUCTION` | Whether the environment is production<br /> | `false` |

1. Per the above table, construct the files listed below. Example code is given for each file.

    `data.tf`
```

As is you're told to build some files, then given key value pairs, then a list of files. Just has a slightly weird flow to it.